system-addon/test/unit/content-src/components/Search.test.js

```diff
@@ -0,0 +1,56 @@
+const React = require("react");
+const {shallow, mount} = require("enzyme");
+const {_unconnected: Search} = require("content-src/components/Search/Search");
+const {actionTypes: at, actionUtils: au} = require("common/Actions.jsm");
+
+const DEFAULT_PROPS = {
+  Search: {
+    currentEngine: {
+      name: "google",
+      icon: "google.jpg"
+    }
+  },
+  dispatch() {}
+};
+
+describe("<Search>", () => {
+  it("should render a Search element", () => {
+    const el = shallow(<Search {...DEFAULT_PROPS} />);
+    assert.ok(el);
+  });
+
+  describe("#performSearch", () => {
+    function clickButtonAndGetAction(el) {
+      const dispatch = el.prop("dispatch");
+      el.find("#search-button").simulate("click");
+      assert.calledOnce(dispatch);
+      return dispatch.firstCall.args[0];
+    }
+    it("should send a SendToMain action with type PERFORM_SEARCH when you click the search button", () => {
+      const el = mount(<Search {...DEFAULT_PROPS} dispatch={sinon.spy()} />);
+      const action = clickButtonAndGetAction(el);
+      assert.propertyVal(action, "type", at.PERFORM_SEARCH);
+      assert.isTrue(au.isSendToMain(action));
```

Comment:
THis and the other tests in this file seem like they would be more readable with the setup/execute/verify structure.